### PR TITLE
Fix phpdoc type errors and review existing explanations of parameters

### PR DESCRIPTION
**Pull request by alexander-zierhut**
- Some type hints are simply wrong or outdated
- Some phpdoc comments are out of date or meaningless

### CODE REVIEW

File: z_model.php [hunk 1]
```diff
@@ -136,14 +136,14 @@ function getResult() {
          * @return int Number of results in the last query
          */
         function countResults() {
-            return $this->z_db->countResults(...func_get_args());
```
Comment (LordVonAdel) [inline z_model.php:139]: Not a docs change

File: z_request.php [hunk 2]
```diff
@@ -471,8 +474,6 @@ public function isAction($type) {
         public function validateCED($name, $rules) {
             $errors = [];
 
-            $result = new FormResult();
```
Comment (LordVonAdel) [inline z_request.php:474]: Where does result come from, when not here?

**Review by LordVonAdel — CHANGES_REQUESTED** [review]

File: z_model.php (hunk 1 above)
Comment (alexander-zierhut) [reply z_model.php:139]: `countResults` does not take any arguments, which I have noticed due to the docs change, `func_get_args` should not be used there as there is nothing to pass.

**Review by LordVonAdel — APPROVED** [review]
Ok

**Pull request merged**